### PR TITLE
Feature: Rework Beneficiaries component, Fix captcha on Login

### PR DESCRIPTION
**Pull request by akhlopiachyi**

### CODE REVIEW

File: src/hooks/useToggle.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export const useToggle = (initialState) => {
```
Comment (oyershov) [inline src/hooks/useToggle.ts:3]: Please remove this hook, it's useless. Use "useState" instead.

**Review by oyershov — CHANGES_REQUESTED** [review]

File: src/components/Beneficiaries/index.tsx [hunk 2]
```diff
@@ -75,242 +43,207 @@ const defaultBeneficiary: Beneficiary = {
...
-            (nextProps.beneficiaries.length && nextProps.beneficiaries !== beneficiaries)) {
-            this.handleSetCurrentAddressOnUpdate(nextProps.beneficiaries);
+            dispatch(memberLevelsFetch());
         }
+    }, []);
+
+    const prevCurrency = usePrevious(currency);
+    const prevBeneficiaries = usePrevious(beneficiaries);
+    const prevBeneficiariesAddSuccess = usePrevious(beneficiariesAddSuccess);
+    const prevBenefiariesActivateSuccess = usePrevious(beneficiariesActivateSuccess);
```
Comment (oyershov) [inline src/components/Beneficiaries/index.tsx:84]: There is no need to have `usePrevious` hook. `useEffect` compares old and new properties.

File: src/components/Beneficiaries/index.tsx [hunk 3]
```diff
@@ -75,242 +43,207 @@ const defaultBeneficiary: Beneficiary = {
...
+            setConfirmationModalState(true);
         }
 
-        if (nextProps.beneficiariesAddSuccess && !beneficiariesAddSuccess) {
-            this.handleToggleAddAddressModal();
-            this.handleToggleConfirmationModal();
+        if (!prevBenefiariesActivateSuccess && beneficiariesActivateSuccess) {
+            setConfirmationModalState(false);
         }
+    }, [beneficiariesAddSuccess, currency, prevBeneficiariesAddSuccess, prevBenefiariesActivateSuccess, beneficiariesActivateSuccess]);
```
Comment (oyershov) [inline src/components/Beneficiaries/index.tsx:100]: 1. Separate this hook to multiple `useEffect()`
2. Each hook should do one action. Example:
**This code**
```
React.useEffect(() => {
        if ((props.currency && prevCurrency !== props.currency)
            || (beneficiaries.length && beneficiaries !== prevBeneficiaries)) {
            handleSetCurrentAddressOnUpdate(beneficiaries);
        }
        ...
```
**Should be like this**
```
 React.useEffect(() => {
        if (beneficiaries) {
            handleSetCurrentAddressOnUpdate(beneficiaries);
        }
 }, [beneficiaries]);
```

**Review by oyershov — CHANGES_REQUESTED** [review]

**Review by oyershov — APPROVED** [review]
LGTM

File: src/translations/en.ts [hunk 4]
```diff
@@ -866,6 +866,7 @@ export const en = {
 
     'account.beneficiary.invalid_pin': 'Invalid beneficiary activation pin',
     'account.beneficiary.cant_resend_within_1_minute': 'Can\'t resend activation pin within 1 minute',
+    'account.beneficiary.duplicate_address': 'Beneficiary address already existed',
```
Comment (josadcha) [inline src/translations/en.ts:869]: 'Beneficiary address already **exists**' please

File: src/custom/translations/ru.ts [hunk 5]
```diff
@@ -869,6 +869,7 @@ export const ru: LangType = {
 
     'account.beneficiary.invalid_pin': 'Invalid beneficiary activation pin',
     'account.beneficiary.cant_resend_within_1_minute': 'Can\'t resend activation pin within 1 minute',
+    'account.beneficiary.duplicate_address': 'Beneficiary address already existed',
```
Comment (josadcha) [inline src/custom/translations/ru.ts:872]: 'Beneficiary address already exists' please

**Review by josadcha — CHANGES_REQUESTED** [review]

**Comment by pkucherenk0** [comment]
Tested locally with master Opendax.
1. After the beneficiary created user will fail to resend the pin or activate it, all API requests will be directed to the `0` id beneficiary
Selection_999(2057)
Selection_999(2058)
2. If a user already has a beneficiary, when he visits the withdraw page beneficiary dropdown opens by it self
2.1 If a user will visit the withdraw of the first currecy with a beneficiary and right after that second currency withdrawal, the beneficiary list will contain a mix of beneficiaries from both currencies 
Selection_999(2059)
3. Beneficiary tooltip is behind beneficiary dropdown
Cryptobase: Wallets - Google Chrome_999(001)

File: web/src/containers/Wallets/History.tsx [hunk 6]
```diff
@@ -66,7 +65,7 @@ export class WalletTable extends React.Component<Props> {
         }
     }
 
-    public componentWillReceiveProps(nextProps) {
+    public componentWillReceiveProps(nextProps: Props) {
```
Comment (akushniruk) [inline web/src/containers/Wallets/History.tsx:68]: Let's change it to componentDidUpdate

**Review by akushniruk — APPROVED** [review]

**Pull request merged**